tests/sagemaker/test_sagemaker_deployment_client.py

```diff
@@ -335,10 +335,20 @@ def test_create_deployment_create_sagemaker_and_s3_resources_with_expected_names
     model_environment = sagemaker_client.describe_model(ModelName=model_name)["PrimaryContainer"][
         "Environment"
     ]
-    assert model_environment == {
+    expected_model_environment = {
         "MLFLOW_DEPLOYMENT_FLAVOR_NAME": "python_function",
         "SERVING_ENVIRONMENT": "SageMaker",
     }
+    if os.getenv("http_proxy") is not None:
+        expected_model_environment.update({"http_proxy": os.environ["http_proxy"]})
+
+    if os.getenv("https_proxy") is not None:
+        expected_model_environment.update({"https_proxy": os.environ["https_proxy"]})
+
+    if os.getenv("no_proxy") is not None:
+        expected_model_environment.update({"no_proxy": os.environ["no_proxy"]})
```

Comment:
In order to cover the case  http(s)_proxy setting existing, we should patch the `os.environ` to make it contains "http(s)_proxy" env ?

Reply:
Could you please elaborate on the ask here?

Reply:
Sounds good, and would you suggest duplicating each of the 9 unit test cases, one without the proxies and one with mocked proxies?

Reply:
> duplicating each of the 9 unit test cases

Don't need to duplicate them, but I would suggest use `@pytest.mark.parametrize` on these tests.

Reply:
Updated PR, the unit tests now include parametrized tests to validate scenarios with and without proxies.